apps/client/ui/SortingSelect/SortingSelect.tsx

```diff
@@ -0,0 +1,140 @@
+import type { ComponentPropsWithoutRef } from 'react';
+import React, { useState, useEffect, useRef, KeyboardEvent } from 'react';
+import clsx from 'clsx';
+import { useDispatch, useSelector } from 'react-redux';
+
+import { selectSortOption } from '@/store/MainSlice';
+import { RootState } from '@/store/store';
+import ArrowDownIcon from '@/ui/Icons/ArrowDown/ArrowDown';
+
+type SortingSelectProps = ComponentPropsWithoutRef<'select'> & {
+  options: string[];
+};
+
+const SortingSelect: React.FC<SortingSelectProps> = ({
+  options,
+  className,
+}) => {
+  const dispatch = useDispatch();
+  const [isOpen, setIsOpen] = useState(false);
+  const [focusedIndex, setFocusedIndex] = useState(null);
```

Comment:
it's a reason why you don't have any focused element when you open dropdown - `null` as an initial value - then calculations from line 70 make index 1 after first arrow down click 👌 you can pass 0 as initial value here and reset state to it when you close the dropdown 😁

Reply:
And if I don't want to have the first element on the list focused at the beginning, can I change this initial state to -1? And after selecting an option set the focusedIndex to the index of that selected option? I think this behavior is similar to the HTML select element but correct me if I'm wrong 😃